sway/commands/permit.c

```diff
@@ -54,6 +60,10 @@ static struct feature_policy *get_policy(const char *name) {
 		}
 		list_add(config->feature_policies, policy);
 	}
+	sway_assert(!strcmp(program, policy->program),
+			"!! DANGER !! symlink modified during policy allocation.");
```

Comment:
Also, you should be storing the fully resolved symlink. When we validate the policy we're going to get the fully resolved path from `/proc/[pid]/exe`, so we need to compare against that to get the appropriate policy. All symlink resolution should happen when the policy is parsed and never again.